File: app/src/main/java/com/compuspulse/data/repository/AuthRepository.java
package com.compuspulse.data.repository;

import com.compuspulse.CampusPulseApp;
import com.compuspulse.data.local.dao.UserDao;
import com.compuspulse.data.local.entity.UserEntity;

import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;

public class AuthRepository {

    private final UserDao userDao;
    private final ExecutorService executorService;

    public AuthRepository() {
        userDao = CampusPulseApp.getInstance().getDatabase().userDao();
        executorService = Executors.newSingleThreadExecutor();
    }

    public void register(String username, String password, AuthCallback callback) {
        executorService.execute(() -> {
            UserEntity existing = userDao.getUserByUsername(username);
            if (existing != null) {
                if (callback != null) {
                    callback.onResult(false, "账号已存在");
                }
                return;
            }

            UserEntity user = new UserEntity(
                    username,
                    password,
                    username,
                    System.currentTimeMillis()
            );
            userDao.insert(user);

            if (callback != null) {
                callback.onResult(true, "注册成功");
            }
        });
    }

    public void login(String username, String password, AuthCallback callback) {
        executorService.execute(() -> {
            UserEntity user = userDao.login(username, password);
            if (callback != null) {
                if (user != null) {
                    callback.onResult(true, "登录成功");
                } else {
                    callback.onResult(false, "账号或密码错误");
                }
            }
        });
    }

    public interface AuthCallback {
        void onResult(boolean success, String message);
    }
}

File: app/src/main/java/com/compuspulse/data/repository/AuthRepository.kt
package com.compuspulse.data.repository

import com.compuspulse.data.remote.http.NetworkClient
import com.compuspulse.data.remote.model.ApiResponse
import com.google.gson.JsonElement
import com.google.gson.reflect.TypeToken
import java.lang.reflect.Type

class AuthRepository {

    private val networkClient: NetworkClient = NetworkClient.getInstance()

    suspend fun register(username: String, password: String): Result<Unit> = runCatching {
        networkClient.clearCookies()
        val formParams = hashMapOf(
            "username" to username,
            "password" to password,
            "repassword" to password
        )

        val response: ApiResponse<JsonElement> = networkClient.postForm(
            "user/register",
            null,
            formParams,
            AUTH_RESPONSE_TYPE
        )
        requireAuthSuccess(response)
        requireSessionCookies()
    }

    suspend fun login(username: String, password: String): Result<Unit> = runCatching {
        networkClient.clearCookies()
        val formParams = hashMapOf(
            "username" to username,
            "password" to password
        )

        val response: ApiResponse<JsonElement> = networkClient.postForm(
            "user/login",
            null,
            formParams,
            AUTH_RESPONSE_TYPE
        )
        requireAuthSuccess(response)
        requireSessionCookies()
    }

    suspend fun logout(): Result<Unit> = runCatching {
        try {
            val response: ApiResponse<JsonElement> = networkClient.get(
                "user/logout/json",
                null,
                AUTH_RESPONSE_TYPE
            )
            requireAuthSuccess(response)
        } finally {
            networkClient.clearCookies()
        }
    }

    private fun requireAuthSuccess(response: ApiResponse<JsonElement>?) {
        if (response == null || !response.isSuccess()) {
            val message = response?.getErrorMsg()
            throw RepositoryException(if (message.isNullOrBlank()) "服务器返回错误" else message)
        }
    }

    private fun requireSessionCookies() {
        if (!networkClient.hasSessionCookies()) {
            throw RepositoryException("登录状态保存失败，请重新登录")
        }
    }

    companion object {
        private val AUTH_RESPONSE_TYPE: Type = object : TypeToken<ApiResponse<JsonElement>>() {}.type
    }
}
